lockfile/src/cyclonedx.rs

```diff
@@ -164,23 +164,37 @@ fn from_purl<T: Component>(component: &T) -> anyhow::Result<Package> {
 
 pub struct CycloneDX;
 
+impl CycloneDX {
+    fn process_components<T: Component>(components: Option<&[T]>) -> anyhow::Result<Vec<Package>> {
+        components.map_or(Ok(Vec::new()), |comp| {
+            let component_iter = filter_components(comp);
+            let mut packages = Vec::new();
+            for comp in component_iter {
+                match from_purl(comp) {
+                    Ok(pkg) => packages.push(pkg),
+                    Err(e) => {
+                        if e.is::<UnknownEcosystem>() {
+                            log::warn!("{:?}", e)
+                        } else {
+                            bail!(e)
```

Comment:
I think `bail!` is great if you're using it like `bail!("some reason")`, but only in cases where you can't use `ensure!`. In this case, it probably does make more sense to write `return Err(e.into());`.

Reply:
Agreed. It would be nice if the message provided more information, like which `package_type` it does not recognize and a notice about how that package will be skipped (maybe also include the `purl` entry).